### PR TITLE
Added a feature to create a new spot with correct validations and associations which is ensuring authenticated users can manage their spots

### PR DESCRIPTION
**Pull request by AMBALERT1**
- [x] Created the middleware to handle validation errors 
- [x] Create the route for creating a new spot 
- [x] Ensure that the route is used in your main router

### CODE REVIEW

File: backend/db/seeders/02-spots.js [hunk 1]
```diff
@@ -13,11 +13,39 @@ module.exports = {
...
+      address: '187 North St',
+      city: 'Cary',
+      state: 'NC',
+      country: 'USA',
+      lat: 56.7789,
+      lng: -135.4890,
+      name: 'Home Away From Home',
+      description: 'Looking for a quick getway from the world, then look no further. This wonderful spot is located just a few blocks away from southpoint mall',
+      price: 150.00,
+      previewImage: 'images/image3.jpg',
```
Comment (AnthonyBronca) [inline backend/db/seeders/02-spots.js]: spots don't have previewImages or avgRatings. Reference the dbDiagram

**Review by AnthonyBronca — CHANGES_REQUESTED** [review]

File: backend/db/seeders/02-spots.js (hunk 1 above)
Comment (AMBALERT1) [reply backend/db/seeders/02-spots.js]: i just viewed my dbDiagram and it doesn't have previewImages or AvgRatings but when I navigate to the API its showing PreviewImages and AvgRatings

Comment (AMBALERT1) [reply backend/db/seeders/02-spots.js]: I have remove it from spots

Comment (AMBALERT1) [reply backend/db/seeders/02-spots.js]: It has been removed

**Review by AnthonyBronca — APPROVED** [review]

**Pull request merged**